### ISSUE TITLE
Shipping Module DHL / UPS / FEDEX etc.

### ISSUE THREAD

**Issue by lisandi** [flectrahq/discussion#37]
https://github.com/openlabs/trytond-shipping-dhl-de
https://github.com/openlabs/trytond-shipping-ups
https://github.com/tryton/stock_package_shipping_ups
https://github.com/tryton/sale_supply_drop_shipment
...

It would be great to have something like that in FlectraHQ Community Version

**Comment by flectrahqadmin** [comment]
why to have such separate integrations, if we can do it something like EasyPost or Postmen and cover all providers.

**Comment by lisandi** [comment]
Postmen would be an option as it offers 2 shipper accounts and 20 labels per month which would be enough for an NGO, School or other low budget institutions. Otherwise I would recommend to integrate those 3 major shippers - actually DHL would be already be enough to FlectraHQ

**Comment by berpweb** [comment]
Easypost is a good option... But still people are using only those major providers like DHL, UPS, Fedex...

**Comment by lisandi** [comment]
Tryton has already the code what would be needed and it is GPLv.3 so simply integrate that and it is working better then the option Odoo has themselves in Enterprise

**Comment by Yakulu** [comment]
@lisandi I fear there is a license problem : GPL can not be combined with LGPL without forcing the whole package to be distributed as GPL (when code is integrated), which is not the choice of Flectra authors

**Comment by lisandi** [comment]
Absolut no problem!
https://www.gnu.org/licenses/license-list.html#GPLIncompatibleLicenses
https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility
as the module can be distributed as a separate part to be integrated and that can be GNU GPL v.3 There is no need to have it integrated by default.

This applies also to all other parts

In General and I mentioned that before, especially because of legal cases odoo will bring up it would be actually best to have an overall license AGPL like in Odoo 8. In that case it won't be any problem to integrate the GPL part even in the core. It would also help to integrate all OCA Code which is AGPL as otherwise it could be that ODOO uses there power over AGPL to get up again with legal cases against Flectra as Flectra has no agreement that OCA is not sueing them. ;-)

If I were Flectra I would do what Yuri already started and relicense the complete work as AGPL integrating the LGPL code from ODOO as well as the AGPL code from OCA as well as the one available from other Python 3 GPL resources incl. Tryton.

The resources would be immense bigger than staying with LGPL. Besides that all code brought to public under AGPL would need to get published and that is the biggest benefit of all as ODOO themselves could not use any of that developments besides their own LGPL parts (only those parts) in commercial closed source products.

There is absolutely no need to use LGPL as AGPL would be the much better choice for all - and LGPL would even make people think that Flectra will be soon the same like Fabien is already -> looking for money money and richness only and having the community only to build up his imperium.

Yuri with Cubic ERP is going the AGPL way and by combining the power with YURI (South America) also the reach could be extended a lot.

Actually, the worst thing what could happen for Odoo is that Flectra goes to be AGPL as the overall License. The Flectra developers can change the license even to AGPL as they did modifications and improvements already in the code. Compare it to an AGPL pillow case you pull over the LGPL pillow itself which means Flectra on the Outside but using Odoo on the inside is 100% OK under AGPL and even under GPLv.3

Unfortunately, GPLv.3 would bear the risk that Odoo buys people with their money who undermine the GPL (hidden development) which is simply not so possible under AGPL license where the code has to be made public accessible as soon as parts of it go online.

The other risk actually Flectra is facing by keeping everything LGPL is that a group of people will take the code and design the AGPL pillow I mentioned! And this will for sure happen even with the power of Indian developers as also here AGPL is the first priority and not LGPL!

**Comment by Yakulu** [comment]
Yes, my comment is about inclusion per default. I know these licenses are compatible.
I agree about AGPL for all, but Flectra authors have been quite clear about that (on Gitter IIRC), they want to allow closed source software on some future app store, and if it is economically needed, they want to be able to have an Enterprise Edition too.

**Comment by parthivgls** [comment]
@lisandi @Yakulu thanks for your input, at the moment we are running with limited resources. 

If I have to spend 40 developer hours, I need to decide whether I should spend it relicensing existing work to LGPL to AGPL or to build a new feature that can be ported along with. 

I personally choose the second one. I know the discussion is always endless about AGPL vs LGPL. 

My only point of concern is not to look back, move on with what's in hand and keep on building. 

We can either make additional module which is AGPL, created from GPL based module and introduce a licensing hell and debates, or we just simply create LGPL based module from scratch and keep in core flectra.

**Comment by lisandi** [comment]
@parthivgls with AGPL you will unite the community and will pull most of the guys immediately over, with LGPL this won't happen as their fear would be the same as with Odoo S.A. and those who like to go the AGPL way incl people from OCA who woke up (unfortunately only some) would rather spend time to develop an AGPL PillowCase where the rest can be packed in in the way you already mentioned.

OCA and Yuri and some others simply won't go LGPL way but would be more than willing to go the AGPL way together I would say.

Besides that, you will have a general problem when you integrate i.e. AGPL OCA code as in the moment you will distribute that as a unite you would violate the licenses like odoo is already doing, but they have an agreement with OCA that they don't get sued. Pretty similar to what Hitler did actually before he tried to take over the world and you know how it ended.

Finally, the world took over Germany ;-). Right now we are in the face that we need to form Allies against the dark side! Let's say it like that and if this is successful The force will be with us - and of course in AGPL.

The point is actually the overall cover of the whole and not the single code line. So if you need code which needs to be integrated into commercial software you still can develop that code as LGPL and it gets integrated into the AGPL pillow which secures it from being taken over by Odoo in the darkness. On the other side, all code odoo would use would need to be AGPL besides their own LGPL code which gives them only problems already.

Not looking back is OK but you should learn out of what happened in the past and why the situation is actually like it is. The problem is the LGPL license and the fear of developers that their contributions get finally be used/abused for purposes they have no access anymore. That was the reason why actually after GPL AGPL had been created. So if you don't want to look back then the only way would be going AGPL 3 as LGPL is a huge step backward and as mentioned already it will only be a matter of some months and you will see an AGPL version, which means another ODO/Flectra Clone in that case which will be used to gather the odoo Community.

I think from the point of Fabien I actually would try to buy Flectra, simply to stop that horror scenario for him as he knows much to well that if Flectra stays LGPL he will have to fight numerous AGPL clones of both Odoo and Flectra. On the other hand he also knows that an AGPL Version of Flectra would rob him the complete base incl OCA as all of their modules suddenly would get developed further for Flectra and as there is a difference in code Fabien would not even be able to reuse those new python 3 code parts for his Odoo. 

No matter what way Fabien for sure knows that he has already lost the race and the only way actually is to buy the complete stuff and make you a companion of Odoo. 

Even worse for him, when you will release Flectra in 2 days on 25th everyone in Brussels will be sitting at the Christmas table for 2 days and then the weekend is near already and new year and school holidays etc It is the best time to win a race against a European company located in a Christian society.

We here in Thailand are working like you in India and elsewhere in the Asian and Arabic world.

The work to AGPL is actually not much as Fabien already pointed out himself by his legal department. They know that AGPL is possible and so they have only the chance that they need to be credited in the code (not in the frontend) well that is absolute no problem and of course you need to create a new work as a cover of all AGPL. but that is the easiest part as you already brought in many code cleanups and modifications which would justify a legal code change besides simply mentioning that the origin of the code came from odoo. Who cares. Soon people won't mention Odoo anymore as a great system but as the last version of a piece of software which lost its community.
